compiler/rustc_mir/src/interpret/validity.rs

```diff
@@ -634,8 +623,8 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
         scalar_layout: &ScalarAbi,
     ) -> InterpResult<'tcx> {
         let value = self.read_scalar(op)?;
-        let valid_range = &scalar_layout.valid_range;
-        let (lo, hi) = valid_range.clone().into_inner();
+        let valid_range = scalar_layout.valid_range;
+        let AllocationRange { start: lo, end: hi } = valid_range;
         // Determine the allowed range
         // `max_hi` is as big as the size fits
         let max_hi = u128::MAX >> (128 - op.layout.size.bits());
```

Comment:
The check being done a few lines down here (testing whether the range is equivalent to the full range containing all values) also sounds like something that could be provided as a method on the new range type.

Reply:
I have thought about it, only problem I see is that we have to compute `max_hi` which is dependent on not just the range, but the allowed number of bits.

Reply:
Yes, that helper function will have to take the `Size` of the scalar.